app/controllers/users_email_controller.rb

```diff
@@ -61,125 +53,55 @@ def update
   end
 
   def confirm_new_email
-    load_change_request(:new)
-
-    if @change_request&.change_state != EmailChangeRequest.states[:authorizing_new]
-      @error = I18n.t("change_email.already_done")
-    end
+    change_request = load_change_request(:new)
 
-    redirect_url = path("/u/confirm-new-email/#{params[:token]}")
+    result =
+      run_second_factor!(SecondFactor::Actions::ConfirmEmail, target_user: change_request.user)
```

Comment:
It looks like `@change_request` could have been null in the previous implementation. Do we need to handle that here as well?

Reply:
I updated the implementation of `load_change_request` so that it raises an exception rather than returniing `nil`, so it should be safe to assume it exists now